homeassistant/components/litterrobot/select.py

```diff
@@ -10,7 +17,59 @@
 from .entity import LitterRobotConfigEntity
 from .hub import LitterRobotHub
 
-TYPE_CLEAN_CYCLE_WAIT_TIME_MINUTES = "Clean Cycle Wait Time Minutes"
+
+@dataclass
+class RobotSelectEntityDescription(SelectEntityDescription):
+    """A class that describes robot select entities."""
+
+    current_fn: Callable[[Robot], int | float | None] = lambda _: None
+    options_fn: Callable[[Robot], list] = lambda _: []
+    select_fn: Callable[
+        [Robot], Callable[..., Coroutine[Any, Any, bool]] | None
+    ] = lambda _: None
+    cast_type: type[int | float] = field(default=int)
```

Comment:
I am not sure if it would work, but maybe you need a double-generic to handle cast_type separately from _RobotT.
ie. `Generic[_RobotT, _CastTypeT]`

You could then also set `options_fn` to return `list[_CastTypeT]`

Reply:
I was just about to ask this question as I haven't been having luck here. The problem I'm then running into is you can't instantiate something of TypeVar as it is only for type hinting, or so the errors seem to suggest

Reply:
I've run out of time today. I'll try to have another look tomorrow if you still have difficulty..

Reply:
Also, maybe @cdce8p has an idea to implement this in the meantime